app/models/effective/post.rb

```diff
@@ -1,5 +1,26 @@
 module Effective
   class Post < ActiveRecord::Base
+    if defined?(PgSearch)
+      include PgSearch::Model
+
+      multisearchable against: [
+                        :title,
+                        :description,
+                        :slug,
+                      ],
+                      associated_against: {
+                        rich_texts: [:body],
+                      },
+                      using: {
+                        trigram: {},
+                        tsearch: {
+                          highlight: true,
+                        }
+                      },
+                      if: -> (post) { !post.draft }
```

Comment:
could you use `post.published?` which is defined below as

```

    def published?
      !draft? && published_at.present? && published_at < Time.zone.now
    end
```

so people can publish stuff in the future, or every wednesday.